sale_exception_max_allowed_discount/model/sale.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Eficent (<http://www.eficent.com/>)
+#              <contact@eficent.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import datetime
+from openerp import models, fields, api
+from openerp.tools.translate import _
+
+
+class SaleOrderLine(models.Model):
+
+    """Adds an exception function to be called by the sale_exceptions module.
+
+    The function will ensure that the discount of the line does not exceed
+    the maximum discount defined for the product.
+
+    """
+
+    _inherit = "sale.order.line"
+
+    @api.multi
+    def _should_skip_discount_checks(self):
+        self.ensure_one()
+
+        if (
+            self.product_id and
```

Comment:
Evaluate this expression with not and assign it to the variable instead of making an if expression.